test/e2e-tests-upgrade.sh

```diff
@@ -16,19 +16,35 @@
 
 # This script calls out to scripts in tektoncd/plumbing to setup a cluster
 # and deploy Tekton Pipelines to it for running upgrading tests. There are
-# two scenarios we need to cover in this script:
+# three scenarios we need to cover in this script:
 
 # Scenario 1: install the previous release, upgrade to the current release, and
 # validate whether the Tekton pipeline works.
 
 # Scenario 2: install the previous release, create the pipelines and tasks, upgrade
 # to the current release, and validate whether the Tekton pipeline works.
 
+# Scenario 3: install the current pipelines, use the integration tests from the previous release(s)
+# to prevent regression. Incompatible changes made in the current release cycle are expected to fail
+# and will be skipped after failing once.
+
 source $(git rev-parse --show-toplevel)/test/e2e-common.sh
 RELEASES='https://github.com/tektoncd/pipeline/releases/latest'
 PREVIOUS_PIPELINE_VERSION=$(curl -L -s -H 'Accept: application/json' $RELEASES |
   sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
 
+# BACKWARD_INCOMPATIBLE_TESTS is a manually maintained list of e2e tests that contains known breaking changes
+# made in the current devel cycle. It prevents the upgrade test job from keeping failing after it failed once.
+# TODO: #5782 this struct could be changed to a map once we are testing against more releases
```

Comment:
This comment is a bit confusing to me, and I think this commit is intended to close #5782. Is the intention to test against multiple older releases, the latest LTS release (as in the lower comment), or both? Is there a better issue that could track this work?

Reply:
I have added https://github.com/tektoncd/pipeline/issues/6728 for the new issue as a subtask to track.